pkg/apis/pipeline/v1alpha1/verificationpolicy_validation_test.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2022 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/test/diff"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+)
+
+func TestVerificationPolicy_Invalid(t *testing.T) {
+	tests := []struct {
+		name               string
+		verificationPolicy *v1alpha1.VerificationPolicy
+		want               *apis.FieldError
+	}{{
+		name: "missing Resources",
+		verificationPolicy: &v1alpha1.VerificationPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "vp",
+			},
+			Spec: v1alpha1.VerificationPolicySpec{
+				Authorities: []v1alpha1.Authority{
+					{
+						Name: "foo",
+						Key: &v1alpha1.KeyRef{
+							Data: "inline_key",
+						},
+					},
+				},
+			},
+		},
+		want: apis.ErrMissingField("resources"),
+	}, {
+		name: "invalid Resources",
```

Comment:
this is the invalid pattern case @bobcatfish

Reply:
perhaps name the case "invalid pattern" to make that clear

Reply:
I changed it to invalid ResourcePattern to match the type name